qa/evil-tests/src/test/java/org/elasticsearch/bootstrap/EvilBootstrapChecksTests.java

```diff
@@ -49,7 +50,17 @@ public void tearDown() throws Exception {
 
     public void testEnforceBootstrapChecks() throws NodeValidationException {
         setEsEnforceBootstrapChecks("true");
-        final List<BootstrapCheck> checks = Collections.singletonList(context -> BootstrapCheck.BootstrapCheckResult.failure("error"));
+        final List<BootstrapCheck> checks = Collections.singletonList(
+            new BootstrapCheck() {
+                @Override
+                public BootstrapCheckResult check(BootstrapContext context) {
+                    return BootstrapCheck.BootstrapCheckResult.failure("error");
+                }
+
+                @Override public ReferenceDocs referenceDocs() {
+                    return null;
```

Comment:
Maybe avoid nulls in tests too just for ease of reading - `ReferenceDocs.BOOTSTRAP_CHECKS` would be a reasonable value to return here too.